routes/transaction.go

```diff
@@ -626,6 +777,238 @@ func GetBalanceForPublicKeyUsingUtxoView(
+		[]*lib.MsgDeSoTxn{basicTransferTxn, updateProfileTxn}, nil, fes.backendServer.GetMempool())
+	if err != nil {
+		_AddBadRequestError(ww,
+			fmt.Sprintf("SubsidizedUpdateProfile: Problem constructing atomic transaction wrapper: %v", err))
+		return
+	}
+
+	// (4) Sign the basic transfer transaction within the atomic transaction wrapper.
+	basicTransferSignature, err :=
+		atomicTxnWrapper.TxnMeta.(*lib.AtomicTxnsWrapperMetadata).Txns[0].Sign(subsidizationPrivateKey)
+	if err != nil {
+		_AddBadRequestError(ww,
+			fmt.Sprintf("SubsidizedUpdateProfile: Problem signing basic transfer transaction: %v", err))
+		return
+	}
+	atomicTxnWrapper.TxnMeta.(*lib.AtomicTxnsWrapperMetadata).Txns[0].Signature.SetSignature(basicTransferSignature)
+
+	// (5) Validate the atomic transaction wrapper has sufficient fees paid.
+	atomicTxnBytes, err := atomicTxnWrapper.ToBytes(true)
+	if err != nil {
+		_AddBadRequestError(ww,
+			fmt.Sprintf("SubsidizedUpdateProfile: Problem serializing transaction: %v", err))
+		return
+	}
+	txnSizeBytes := uint64(len(atomicTxnBytes))
+	if txnSizeBytes != 0 && utxoView.GlobalParamsEntry.MinimumNetworkFeeNanosPerKB != 0 {
+		// Check for overflow or minimum network fee not met.
+		if totalFees != ((totalFees*1000)/1000) ||
+			(totalFees*1000)/uint64(txnSizeBytes) < utxoView.GlobalParamsEntry.MinimumNetworkFeeNanosPerKB {
+			_AddBadRequestError(ww, fmt.Sprint("SubsidizedUpdateProfile: Transactions used to construct"+
+				" atomic transaction do not cumulatively pay sufficient network fees to cover wrapper"))
+			return
+		}
+	}
+
+	// (6) Collect the update profile bytes so the frontend can easily sign it.
+	updateProfileTxnBytes, err :=
+		atomicTxnWrapper.TxnMeta.(*lib.AtomicTxnsWrapperMetadata).Txns[1].ToBytes(true)
+	if err != nil {
+		_AddBadRequestError(ww,
+			fmt.Sprintf("SubsidizedUpdateProfile: Problem serializing update profile transaction: %v", err))
+		return
+	}
+
+	// (7) Return the atomic transaction wrapper to the user.
+	res := SubsidizedUpdateProfileResponse{
+		IncompleteAtomicTransaction:    atomicTxnWrapper,
+		IncompleteAtomicTransactionHex: hex.EncodeToString(atomicTxnBytes),
+		UpdateProfileTransactionHex:    hex.EncodeToString(updateProfileTxnBytes),
+	}
+	if err = json.NewEncoder(ww).Encode(res); err != nil {
+		_AddBadRequestError(ww,
+			fmt.Sprintf("SubsidizedUpdateProfile: Problem encoding response as JSON: %v", err))
+		return
+	}
+}
+
+// ComputeNecessarySubsidyForTransaction is a helper function to find out how much
+// a user must be transferred in order to exactly execute a given transaction.
+//
+// In the balance model era, the total amount of DESO required to execute a transaction
+// is equivalent to the sum of the fees paid and the sum of the basic transfers (outputs).
+func ComputeNecessarySubsidyForTransaction(txn *lib.MsgDeSoTxn) uint64 {
+	necessarySubsidy := txn.TxnFeeNanos
+	for _, txOutput := range txn.TxOutputs {
```

Comment:
We must delete this otherwise the wallet can be drained instantaneously.

Reply:
I wasn't sure this would be accessible to the frontend. Isn't this a risk only if we accept transactions to subsidize from the frontend?

Currently transactions are constructed AND subsidized in the backend in the same endpoint, meaning ComputeNecessarySubsidyForTransaction should be safe. Happy to change it if we want though, could be bad to have this in general.

Reply:
I make a request to `subsidized-update-profile` and add `TransactionFees: [{ PublicKeyBase58Check: 'some-public-key', AmountNanos: 10*1e9}]`. Now in line 875 of your transaction.go file, this gets added as an additional output in `updateProfileTxn`. `updateProfileTxn` is passed into `ComputeNecessarySubsidyForTransaction` and I've just taken 10 DESO from the subsidy wallet.

Reply:
Ahh yeah you're correct... didn't realize the transaction request data enabled the user to specify the transaction fees. Assumed we would compute that on the backend based on network conditions.

Reply:
yeah they aren't network fees really - they allow app developers to easily request a amount of DESO when using their app to transact.

Reply:
Great catch!